Patch taken from Fedora:
http://pkgs.fedoraproject.org/cgit/rpms/dt.git/commit/dt-default-source-define.patch?id=a3107b34a3efc6af66a5f892514f233bb66eb6b5

_BSD_SOURCE was deprecated, hence switch to use _DEFAULT_SOURCE as well
to avoid the build from breaking on newer toolchains.

Signed-off-by: Gustavo Zacarias <gustavo.zacarias@free-electrons.com>

diff -rupN dt.v18.32.old/common.h dt.v18.32/common.h
--- dt.v18.32.old/common.h	2013-03-01 23:52:30.000000000 +0100
+++ dt.v18.32/common.h	2014-09-24 17:41:40.777604710 +0200
@@ -127,7 +127,7 @@ typedef volatile slarge_t	v_slarge;
 
 #endif /* defined(_WIN64) */
 
-#elif defined(__GNUC__) && defined(_BSD_SOURCE) || defined(SCO) || defined(__QNXNTO__) || defined(SOLARIS) || defined(HP_UX) || defined(AIX) || defined(_NT_SOURCE) 
+#elif defined(__GNUC__) && defined(_BSD_SOURCE) || defined (_DEFAULT_SOURCE) ||  defined(SCO) || defined(__QNXNTO__) || defined(SOLARIS) || defined(HP_UX) || defined(AIX) || defined(_NT_SOURCE) 
 
 #define QuadIsLongLong
 typedef unsigned long long int	large_t;
